### PR TITLE
Solution of cafe

### PR DESCRIPTION
**Pull request by svyshcho**

### CODE REVIEW

File: app/cafe.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from datetime import date
+
+from app.errors import NotVaccinatedError,\
+    OutdatedVaccineError,\
+    NotWearingMaskError
```
Comment (Bohdan-Salamakha) [inline app/cafe.py:5]:
```suggestion
from app.errors import (
    NotVaccinatedError,
    OutdatedVaccineError,
    NotWearingMaskError
    )
```

File: app/cafe.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if visitor.get("vaccine") is None:
+            # "vaccine" not in visitor:
```
Comment (Bohdan-Salamakha) [inline app/cafe.py:14]: remove comment

File: app/main.py [hunk 3]
```diff
@@ -1 +1,16 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import NotWearingMaskError, VaccineError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
```
Comment (Bohdan-Salamakha) [inline app/main.py:5]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: app/cafe.py (hunk 1 above)
Comment (svyshcho) [reply app/cafe.py:5]: Done

File: app/cafe.py (hunk 2 above)
Comment (svyshcho) [reply app/cafe.py:14]: Done

File: app/main.py (hunk 3 above)
Comment (svyshcho) [reply app/main.py]: Done

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v2

File: app/main.py [hunk 4]
```diff
@@ -1 +1,16 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import NotWearingMaskError, VaccineError
+
+
+def go_to_cafe(friends: list[Cafe], cafe: Cafe) -> str:
```
[the remark this answers is not in the document]
Comment (svyshcho) [reply app/main.py]: Confused))
Done!

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**